api/config/dex/config_request.go

```diff
@@ -253,3 +255,12 @@ func checkCertsPEM(cfgErr *shared.InvalidConfigError, key, data string) {
 		}
 	}
 }
+
+func (samlCfg *ConfigRequest_V1_Saml) setNameIDPolicyDefault() {
+	if samlCfg == nil {
+		return
+	}
```

Comment:
What makes this code unique, in that it requires checking for the zero-object of the type? I cannot recall ever seeing this check in all the auth methods we've implemented. (That is, clearly if we got a nil here we would not want to continue. But wouldn't that be true in lots of methods in auth code, too?)

Reply:
Not unique, just a different use case: this is called as
```go
c.V1.Sys.GetConnectors().GetSaml().setNameIDPolicyDefault()
```
and the `nil` check allows us *not* to check if the saml connector is actually set -- but **if it is**, we'll add the default value.